packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -338,7 +346,7 @@ function resetStack() {
   nextUnitOfWork = null;
 }
 
-function commitAllHostEffects() {
+function commitAllHostEffects(root: FiberRoot) {
```

Comment:
Hm. Unfortunate to pass another arg just for this. @acdlite does this look right to you? Can we get this from somewhere else?

Reply:
The chain is ... -> `performWorkOnRoot(root)` -> `completeRoot(root)` -> `commitRoot(root)` -> `commitAllHostEffects(root)`

Why is passing `root` to _this specific function_ a problem? 😄

Reply:
Some functions get inlined so those variables don't matter much.

Extra variables can be *much* worse than closures or property reads. Because if we tip over the number of variables used then the VM will try to guess which ones to pop out of the register and back to main memory.

The key here is that this variable isn't used by the mainline path here. Just the special case path of the profiler.

Reply:
> Should be able to use `nextRoot`, but I assume Brian has passed it as an argument to avoid another type check, since Flow doesn't know that it's not null.

Maybe I'm missing something... `nextRoot` _is_ actually null in many cases though.

`renderRoot()` sets `nextRoot` to null to indicate that there's no more work to be done for the current batch. This means that if `performWorkOnRoot()` then calls `completeRoot()` – the value will be null.

I assume this is why we've been passing around a `root` param to these other functions in the first place.